daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseXMLReader.scala

```diff
@@ -43,35 +41,34 @@ class DaffodilParseXMLReader(dp: DataProcessor) extends DFDL.DaffodilParseXMLRea
   private var errorHandler: ErrorHandler = _
   private var dtdHandler: DTDHandler = _
   private var entityResolver: EntityResolver = _
-  var saxParseResultPropertyValue: ParseResult = _
-  var saxBlobDirectoryPropertyValue: Path = Paths.get(System.getProperty("java.io.tmpdir"))
-  var saxBlobPrefixPropertyValue: String = "daffodil-sax-"
-  var saxBlobSuffixPropertyValue: String = ".blob"
-
-  private val featureMap = mutable.Map[String, Boolean](
-    XMLUtils.SAX_NAMESPACES_FEATURE -> false,
-    XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE -> false
-  )
+  private var saxParseResultPropertyValue: ParseResult = _
+  private var saxBlobDirectoryPropertyValue: Path = Paths.get(System.getProperty("java.io.tmpdir"))
+  private var saxBlobPrefixPropertyValue: String = "daffodil-sax-"
+  private var saxBlobSuffixPropertyValue: String = ".blob"
+  private var saxNamespaceFeatureValue: Boolean = true
+  private var saxNamespacePrefixesFeatureValue: Boolean = false
 
   override def getFeature(name: String): Boolean = {
-    if (name == XMLUtils.SAX_NAMESPACES_FEATURE ||
-      name == XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE) {
-      featureMap(name)
-    } else {
-      throw new SAXNotRecognizedException("Feature unsupported: " + name + ".\n" +
-        "Supported features are: " + featureMap.keys.mkString(", ")
-      )
+    name match {
+      case XMLUtils.SAX_NAMESPACES_FEATURE => saxNamespaceFeatureValue
+      case XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE => saxNamespacePrefixesFeatureValue
+      case _ =>
+        throw new SAXNotRecognizedException("Feature not found: " + name + ".\n" +
+        "Supported features are: " +
+        Seq(XMLUtils.SAX_NAMESPACES_FEATURE,
+          XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE).mkString(", "))
     }
   }
 
   override def setFeature(name: String, value: Boolean): Unit = {
-    if (name == XMLUtils.SAX_NAMESPACES_FEATURE ||
-      name == XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE) {
-      featureMap(name) = value
-    } else {
-      throw new SAXNotRecognizedException("Feature unsupported: " + name + ".\n" +
-        "Supported features are: " + featureMap.keys.mkString(", ")
-      )
+    name match {
+      case XMLUtils.SAX_NAMESPACES_FEATURE => saxNamespaceFeatureValue = value
+      case XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE => saxNamespacePrefixesFeatureValue = value
+      case _ =>
+        throw new SAXNotRecognizedException("Feature unsupported: " + name + ".\n" +
+          "Supported features are: " +
+          Seq(XMLUtils.SAX_NAMESPACES_FEATURE,
+            XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE).mkString(", "))
```

Comment:
Suggest we have a function that throws this exception so we don't have to duplicate this logic.